PXC-3942: Merge PS-5.7.38 Part-2

venkatesh-prasad-v opened the pull request:
Implemented PXC-3942 (Merge PS-5.7.38, Part 2 no conflicts)

https://jira.percona.com/browse/PXC-3942

Percona Server release 5.7.38-41

Merge tag 'Percona-Server-5.7.38-41' into merge5738-part2

Testing Done
----
**Jenkins**
- galera suites:https://pxc.cd.percona.com/view/PXC%205.7/job/pxc-5.7-param/308/console
Failing tests: None

- Full MTR: https://pxc.cd.percona.com/view/PXC%205.7/job/pxc-5.7-param/309/console
Failing tests: Failing tests: `main.merge` and `main.merge_mmap`. These are known failures. https://jira.percona.com/browse/PXC-3296

lipengchong committed:
PS-8129: Update threadpool_unix.cc

if io_poll_create failed ,we shouldn't break.we should unlock the group mutex and return i. if  break here,we don't unlock the group mutex,and
group_count= size.So it is a bug here.

bkandasa committed:
Raise version number after cloning 5.6.55

Approved-by: Bjorn Munch <bjorn.munch@oracle.com>

bkandasa committed:
Raise version number after cloning 5.7.37

Approved-by: Bjorn Munch <bjorn.munch@oracle.com>

sachinagarwal1111 committed:
Bug #33101844 CREATE FULLTEXT INDEX CRASHES SERVER FOR LARGE TABLE

Problem:
When an Index is added to the table. Index entries are
first sorted & merged and then index tree is created in bottom up
order to expedite the processing. while creating index tree,
X-latch is taken on the rightmost page at each level and kept till
the end of index creation. meanwhile if page cleaner thread chooses
any of the non-leaf page (or root page) from index tree for flush
then page cleaner thread keeps waiting for X-latch to be released
from that page and eventually hits fatal timeout error.

Fix:
During index creation, X-latch is taken on the non-leaf page
only for insert into that page and release immediately.

RB: 27333
Reviewed by : Annamalai Gurusami <annamalai.gurusami@oracle.com>

MRajcic committed:
Bug#33619511: MySQL server exits in debug build when -p parameter used

Fixes server exit due to uninitialized variables in keyring migration code.

Approved by: Harin Vadodaria <harin.vadodaria@oracle.com>

Change-Id: I05bf3dfd059140c4c1ac0e14a9260e51f5a40ca8

nacarvalho committed:
BUG#33581529: Backport Fix for Bug 29223844 to 5.7 version of Group Replication

An OOM can happen on transaction marshalling to the network in the
group_replication_trans_before_commit() hook.

To solve the above issue that situation is handled resulting in the
transaction rollback.

RB: 27439

MRajcic committed:
Bug#33619511: MySQL server exits in debug build when -p parameter used

post-push fix: Fix for mysql-5.7 using C++03 standard

Change-Id: I6efb2974a4222a7fcd0423bc16ae510357f5f215

MRajcic committed:
Bug#33619511: MySQL server exits in debug build when -p parameter used

post-push fix: Removing test in MySQL 5.7 only, it behaves differently compared to 8.0

Change-Id: I401e6ecf07d13b03abe09815a9cc22c9464cf6e6

harinvadodaria committed:
Bug#33578113: DROP privilege on performance_schema.* can't be revoked

Description: Just like GRANT on PFS and tables within
             it is allowed, REVOKE should be allowed
             too.

Fix: Updated PFS_internal_schema_access::check() to
     allow privilege checks to proceed as usual for
     SQLCOM_REVOKE when privilege in question is
     DROP_ACL.

RB: 27429

Commit:
Bug #33659435 Bundled libevent: EVENT__HAVE_WORKING_KQUEUE not defined with Xcode 12 or later

Apply upstream patch: CheckWorkingKqueue.cmake: fix missing headers

Avoid implicitly declaring functions memset() and exit()
as it is considered an error by some compilers
(e.g. Xcode clang 12 and later)

Change-Id: I6d302ec4c4d08d86e91ebadf78007ad05d58112b
(cherry picked from commit 277b62d0933ef925ffdf499c77735abbc0972fb3)
(cherry picked from commit c1e96e8dffb760834722547a4ea9cfbc35d3d5ec)

Commit:
Bug #33668341 MySQL 5.7: Look for openssl@1.1 on Mac

Homebrew has added support for OpenSSL 3.0
This is currently not supported by our codebase,
so look explicitly for openssl 1.1

Change-Id: I3856cf1830ab14d32317f864e4b0122dcb2371c1

salmankhan123417 committed:
BUG#30042589 MYSQLDUMP LEAKS MEMORY WHEN SELECTED TABLES ARE DUMPED WITH --ORDER-BY-PRIMARY

Description: Executing mysqldump on more than 1 table with
--order-by-primary option leaks memory.

Analysis: Dump_table() is invoked once for each table when executing
dump_selected_tables, So memory for global order_by is allocated
every time. But while freeing its done only once and the previously
allocated memory gets leaked. Memory for order_by is allocated when
get_table_structure() is being called.

Fix: Order_by is freed within dump_table() thus freeing it for every
invocation and also its freed after every get_table_structure() call.

RB: 27465

gkodinov committed:
WL#14937: disable multi-threaded myisam repair in 8.0
RB#27440

The 5.7 version.

Commit:
Bug#33334507 - 5.7 - utf8mb4 + REPLACE(UUID(),'-','') returns
duplicate UUIDs!

Item_func_uuid gets improperly cached and hence returns the same
value for subsequent calls due to missing flag.

Fix:
Set RAND_TABLE_BIT flag to Item_func_uuid so as to be considered
non-constant and further on: non-cacheable.

Change-Id: I772aa15e91d08e5590b24af4bfcc2c6d2428157e

Commit:
BUG #33459653 : coredump after add index for partion table with ibd
file missing
Problem:
- If we delete .ibd file of a partition along with the 1st partition
  then after restart when we open the partition table we get the
  error message, which is the expected behaviour.
- When we delete .ibd file of a partition other than the 1st
  partition then after restart when we open the partition table we
  hit the debug assert because ha_innopart::open() checks for
  ibd_file_missing only for the 1st partition.

Solution:
Instead of checking ibd_file_missing only for the 1st
partition we check ibd_file_missing for all the partitions.

Reviewed by : Marcin Babij (marcin.babij@oracle.com)
RB  #27494

Commit:
Post push fix

Reviewed by : Marcin Babij (marcin.babij@oracle.com)
RB  #27494

sachinagarwal1111 committed:
Bug #33600109 Segmentation fault at ha_innobase::index_type，
Improper code handling

Problem:
The server may terminates abruptly because nullptr check is
missing for index instance.  The index object is returned by
innobase_get_index() and the return value may be 'nullptr'.
This means that the input parameter of the function
dict_index_is_spatial() is nullptr. That will cause the server
either hits a debug assert or reports a segmentation fault.

Fix:
Added a nullptr check for index instance.

RB: 27517
Reviewed by : Debarun Banerjee <debarun.banerjee@oracle.com>

nacarvalho committed:
WL#15060: Change group_replication_transaction_size_limit default in 5.7

The option `group_replication_transaction_size_limit` limits the
size of transactions allowed in a Group Replication primary, that
is, if a given transaction exceeds this value it will be rollback.
This option was added in 5.7.19 and 8.0.2-dmr, since 5.7 was a GA
version it was added with a default value of 0 to avoid behaviour
change. On 8.0 the default value is 150000000.

When its value is 0 the memory used for transactions is unbounded,
which can cause out of memory errors.
To prevent these errors we did change the 5.7 default to
150000000, the already default value on 8.0.

RB: 27623

Commit:
Bug 26906787 - INCORRECT AUTO_INCREMENT IDS ISSUED WHEN CLOSE TO ULLONG_MAX

PROBLEM
-------
1) ha_innobase::get_auto_increment() is supposed to return the previous
   auto inc value if   the increment causes overflow ( i.e maximum size of the column)
2) We were not recognizing the maximum value of the column and proceeding
   to calculate the auto inc value which causes wrong values to be generated.
FIX
---
1) Recognize if the increment causes overflow and return the previous auto inc value

#rb 27654
Reviewed-by : Debarun Banerjee <debarun.banerjee@oracle.com>

trosten committed:
Bug#33903639 Fix spell of log messages in mysql_safe

 Fix some spelling mistakes in mysqld_safe.
 Based on contribution from bin wang.

Change-Id: Ia4a7e99f3b42d754dd7db6c5470381e41ec9d368

Commit:
BUG #32586721 : SERVER CRASH IN FIL_SPACE_IS_BEING_TRUNCATED

Problem:
- If we start server with keyring and delete record from encrypted table,
  keeping innodb_purge_stop_now=ON.
- On restarting the server, purge threads will try to process undo records
  of encrypted table. However tablespace for encrypted table will not be
  loaded.
- This will result in termination of the server.

Fix:
  Skip undo records for which tablespace is missing.

Reviewed by : Annamalai Gurusami (annamalai.gurusami@oracle.com)
RB  #27559

Commit:
Bug#33732907: Slow query log logs password in plain text on syntax error

If a statement could not be parsed, it would not be included in
the general log (unless --log-raw was used, in which case the
statement would be logged before parsing).

By contrast, the slow query log would include such a query,
which in theory could expose sensitive credentials.

This patch changes the behaviour so that unparseable queries are
not slow-logged. This aligns it with the default behaviour of
the general log. The slow query log will not observe --log-raw
or align its behaviour with the performance schema's, as it is
not the correct tool to debug malformed queries or hunt down
possible attempts at SQL injection.

Backport from 8.x

Change-Id: I2395bc853b8c66a60ab6e1578dceffbb48de2c05

bkandasa committed:
Update License Book

Approved-by: Bjorn Munch <bjorn.munch@oracle.com>

dutow committed:
Merge pull request #1809 from lipengchong/5.7

PS-8129: bug in  threadpool_unix.cc  tp_set_threadpool_size() (5.7)

Commit:
PS-6029 - Fix format of result of data_masking.gen_rnd_us_phone

https://jira.percona.com/browse/PS-6029

Commit:
PS-6029 - Use file suite.opt in the suite data_masking

https://jira.percona.com/browse/PS-6029

It allows a more succinct implementation of the test suite.

ldonoso committed:
Merge pull request #4692 from ldonoso/PS-6029-5.7

PS-6029 - Fix format of result of data_masking.gen_rnd_us_phone

altmannmarcelo committed:
Fixed PS-8136 - LOCK TABLES FOR BACKUP is not preventing ALTER INSTANCE ROTATE INNODB MASTER KEY

https://jira.percona.com/browse/PS-8136

Problem:
Lightweight MDL LTFB should block operations that will result on
inconsistent backups. Rotating master key after the backup has started
will make tables header to point to a key that is not present on backup
cache of keys as PXB reads them once at the beginning of the backup.
Rotating the master key should be blocked.

Fix:
Enhanced Rotate_innodb_master_key to wait for up to lock_wait_timeout
seconds in case other session has acquired LTFB and also fail in case
LTFB has been acquired by the current session.

Commit:
PS-7563 - Fix read overflow in mysqltest

https://jira.percona.com/browse/PS-7563

*Problem:*

`buf` might not contain a final `\0` so, when `buf` is passed to `DBUG_PRINT`,
`DBUG_PRINT` will read pass over the limits of `buff` till if finds a
`\0`.

This is reported by AddrssessSanitizer has invalid access to
memory:

```
 cd $(BUILD_DIR_57)/mysql-test && \
 ./mtr \
 --parallel=1 \
 --force \
 --max-test-fail=0 \
 --retry-failure=0 \
 --suite-timeout=9999 \
 --debug-common \
 --sanitize \
 binlog_encryption.binlog_index

...

=================================================================
==314051==ERROR: AddressSanitizer: stack-buffer-overflow on address
0x7ffca6fa5d00 at pc 0x000000434159 bp 0x7ffca6fa5210 sp 0x7ffca6fa49d0
READ of size 258 at 0x7ffca6fa5d00 thread T0
    #0 0x434158 in strnlen
       (/home/ldonoso/src/bld57/client/mysqltest+0x434158)
    percona#1 0x52bcc1 in process_str_arg
       /home/ldonoso/src/percona-server/strings/my_vsnprintf.c:206:9
    percona#2 0x5297a2 in my_vsnprintf_ex
       /home/ldonoso/src/percona-server/strings/my_vsnprintf.c:619:11
    percona#3 0x52c1fb in my_vsnprintf
       /home/ldonoso/src/percona-server/strings/my_vsnprintf.c:682:10
    percona#4 0x571a6f in DbugVfprintf
       /home/ldonoso/src/percona-server/dbug/dbug.c:1498:10
    percona#5 0x57181f in _db_doprnt_
       /home/ldonoso/src/percona-server/dbug/dbug.c:1432:3
    percona#6 0x4f5482 in LogFile::show_tail(unsigned int)
       /home/ldonoso/src/percona-server/client/mysqltest.cc:808:7
    percona#7 0x4cc8b4 in die(char const*, ...)
       /home/ldonoso/src/percona-server/client/mysqltest.cc:1559:12
    percona#8 0x4ed694 in main
       /home/ldonoso/src/percona-server/client/mysqltest.cc:9915:9
    percona#9 0x7f9bff93d0b2 in __libc_start_main
       /build/glibc-eX1tMB/glibc-2.31/csu/../csu/libc-start.c:308:16
    percona#10 0x42125d in _start
        (/home/ldonoso/src/bld57/client/mysqltest+0x42125d)

Address 0x7ffca6fa5d00 is located in stack of thread T0 at offset 352 in
frame
    #0 0x4f51df in LogFile::show_tail(unsigned int)
       /home/ldonoso/src/percona-server/client/mysqltest.cc:772

  This frame has 2 object(s):
    [32, 64) '_db_stack_frame_' (line 773)
    [96, 352) 'buf' (line 783) <== Memory access at offset 352 overflows
this variable

```

*Solution:*

Use `%.*s` so at most `bytes` are handed over to `DBUG_PRINT`.

Commit:
PS-7769 - Fix use-after-return error in audit_log_exclude_accounts_validate

---

*Problem:*

`st_mysql_value::val_str` might return a pointer to `buf` which after
the function called is deleted. Therefore the value in `save`, after
reuturnin from the function, is invalid.

In this particular case, the error is not manifesting as val_str`
returns memory allocated with `thd_strmake` and it does not use `buf`.

*Solution:*

Allocate memory with `thd_strmake` so the memory in `save` is not local.

EvgeniyPatlan committed:
Merge pull request #4700 from percona/release-5.7.37-40

Release 5.7.37 40

patrickbirch committed:
PS-8099 Document rocksd_enable_native_partitioning (5.7) (#4672)

M  source/myrocks/variables.rst

Co-authored-by: Alina Derkach <81975178+alina-derkach-oaza@users.noreply.github.com>

patrickbirch committed:
PS-7055 Correct PS 5.7 Install Instructions (#3842)

Modified apt_repo
added section on AppArmor
updated link for debian pin

patrickbirch committed:
PS-8101 Documentation for Thread Pool functionality in 5.7 has duplicated text (5.7) (#4670)

M source/performance/threadpool.rst

patrickbirch committed:
PS-8140 Update MyRocks variable description (5.7) (#4694)

M  source/myrocks/variables.rst

patrickbirch committed:
PS-8121 Document do not use dot(.) on binlog log_bin and log_bin_index variables (5.7) (#4683)

M source/flexibility/binlogging_replication_improvements.rst

patrickbirch committed:
PS-8107 Audit log option audit_log_rotate_on_size does not specify unit of measure (5.7) (#4671)

M source/management/audit_log_plugin.rst

patrickbirch committed:
PS-8100 Documentation note about TLSv1.3 support for RHEL 7 (5.7) (#4667)

M source/installation/yum_repo.rst

patrickbirch committed:
PS-8094 rocksdb_bulk_load variable description is misleading (5.7) (#4652)

M source/myrocks/variables.rst

Co-authored-by: Alina Derkach <81975178+alina-derkach-oaza@users.noreply.github.com>

patrickbirch committed:
PS-8093 Wrong documentation for rocksdb_perf_context_level (5.7) (#4651)

M source/myrocks/variables.rst

patrickbirch committed:
PS-8054 Typos on doc pages "Installing Percona Server for MySQL on Debian and Ubuntu" 5.7 (#4606)

M  source/installation/apt_repo.rst

patrickbirch committed:
PS-3740 LP #1716507: innodb_flush_method needs updating in documentation (#4216)

modified files:
scalability/innodb_io
changed the default value
changed the allowed values to match 5.6

alina-derkach-oaza committed:
PS-8069 Remove extra Data masking section from the table of contents modified: source/index.rst (#4628)

Co-authored-by: “Alina <“alina.derkach@percona.com”>

alina-derkach-oaza committed:
PS-8084 Create 5.7.37-40 release notes and release note tasks 	modified:   source/conf.py (#4647)

new file:   source/release-notes/Percona-Server-5.7.37-40.rst
	modified:   source/release-notes/release-notes_index.rst

Co-authored-by: “Alina <“alina.derkach@percona.com”>

ldonoso committed:
Merge pull request #4698 from ldonoso/PS-7563-5.7

PS-7563 - Fix read overflow in mysqltest

Commit:
Add grep and ps to lsan.supp

*Problem*:

Builds using `WITH_ASAN=ON` present the following issues in some platforms:

- `have_grep.inc` fails in platforms which do have `grep`.
- The following 2 tests hang forever: `main.mysqld_safe` and `main.percona_processlist_tid`

The error is caused because when `LD_PRELOAD` is set to include the ASAN library. This messes up with utilities as `grep` and `ps`. In some platforms, memory leaks are reported to be detected on those utilites, but in some other platforms an error not related with memory leaks is returned.

In the case of `grep` an error is returned when testing its presence in the platform, hence `have_grep.inc` interprets `grep` is not installed.

The circumstances under `LD_PRELOAD` is set by mtr are:

- gcc compiler is used.
- package tirpc installed.

The logic implementing this is located in <https://github.com/percona/percona-server/blob/0e89ba12cf9a1098e3e4c800449072727c1e6e37/mysql-test/lib/My/SafeProcess/CMakeLists.txt#L47-L76>

*Solution*:

Add the problematic programs to `lsan.supp`. Although, this solves just the problem of these programs failing because of memory leaks.

Commit:
PS-8145: Fix MYSQL_LEX_CSTRING initialization in sql_audit::mysql_audit_notify()

https://jira.percona.com/browse/PS-8145

The sizeof(const char *) was used as a string length during general_sql_command
initialization mysql_event_general_subclass_t audit event. As a result
there was wrong string length in some cases.
Changed it to use std::string::length() to properly initialize string
length.

ldonoso committed:
Merge pull request #4703 from ldonoso/PS-7848-5.7

Add grep and ps to lsan.supp

Commit:
Fix test main.bug12969156 when WITH_ASAN=ON

*Problem:*

ASAN complains about stack-buffer-overflow on function `mysql_heartbeat`:

```
==90890==ERROR: AddressSanitizer: stack-buffer-overflow on address 0x7fe746d06d14 at pc 0x7fe760f5b017 bp 0x7fe746d06cd0 sp 0x7fe746d06478
WRITE of size 24 at 0x7fe746d06d14 thread T16777215

Address 0x7fe746d06d14 is located in stack of thread T26 at offset 340 in frame
    #0 0x7fe746d0a55c in mysql_heartbeat(void*) /home/yura/ws/percona-server/plugin/daemon_example/daemon_example.cc:62

  This frame has 4 object(s):
    [48, 56) 'result' (line 66)
    [80, 112) '_db_stack_frame_' (line 63)
    [144, 200) 'tm_tmp' (line 67)
    [240, 340) 'buffer' (line 65) <== Memory access at offset 340 overflows this variable
HINT: this may be a false positive if your program uses some custom stack unwind mechanism, swapcontext or vfork
      (longjmp and C++ exceptions *are* supported)
Thread T26 created by T25 here:
    #0 0x7fe760f5f6d5 in __interceptor_pthread_create ../../../../src/libsanitizer/asan/asan_interceptors.cpp:216
    #1 0x557ccbbcb857 in my_thread_create /home/yura/ws/percona-server/mysys/my_thread.c:104
    #2 0x7fe746d0b21a in daemon_example_plugin_init /home/yura/ws/percona-server/plugin/daemon_example/daemon_example.cc:148
    #3 0x557ccb4c69c7 in plugin_initialize /home/yura/ws/percona-server/sql/sql_plugin.cc:1279
    #4 0x557ccb4d19cd in mysql_install_plugin /home/yura/ws/percona-server/sql/sql_plugin.cc:2279
    #5 0x557ccb4d218f in Sql_cmd_install_plugin::execute(THD*) /home/yura/ws/percona-server/sql/sql_plugin.cc:4664
    #6 0x557ccb47695e in mysql_execute_command(THD*, bool) /home/yura/ws/percona-server/sql/sql_parse.cc:5160
    #7 0x557ccb47977c in mysql_parse(THD*, Parser_state*, bool) /home/yura/ws/percona-server/sql/sql_parse.cc:5952
    #8 0x557ccb47b6c2 in dispatch_command(THD*, COM_DATA const*, enum_server_command) /home/yura/ws/percona-server/sql/sql_parse.cc:1544
    #9 0x557ccb47de1d in do_command(THD*) /home/yura/ws/percona-server/sql/sql_parse.cc:1065
    #10 0x557ccb6ac294 in handle_connection /home/yura/ws/percona-server/sql/conn_handler/connection_handler_per_thread.cc:325
    #11 0x557ccbbfabb0 in pfs_spawn_thread /home/yura/ws/percona-server/storage/perfschema/pfs.cc:2198
    #12 0x7fe760ab544f in start_thread nptl/pthread_create.c:473
```

The reason is that `my_thread_cancel` is used to finish the daemon thread. This is not and orderly way of finishing the thread. ASAN does not register the stack variables are not used anymore which generates the error above.

This is a benign error as all the variables are on the stack.

*Solution*:

Finish the thread in orderly way by using a signalling variable.

ldonoso committed:
Merge pull request #4699 from ldonoso/PS-7769-5.7

PS-7769 - Fix use-after-return error in audit_log_exclude_accounts_validate

ldonoso committed:
Merge pull request #4709 from ldonoso/PS-7841-5.7

Fix test main.bug12969156 when WITH_ASAN=ON

Commit:
PS-8150 - Add clang-14 testing to Azure pipelines

Commit:
PS-8151 - Fix warnings when compiling with clang 14

oleksandr-kachan committed:
Merge pull request #4704 from oleksandr-kachan/PS-8145-5.7

PS-8145: Fix MYSQL_LEX_CSTRING initialization in sql_audit::mysql_audit_notify() (5.7)

ldonoso committed:
Merge pull request #4716 from ldonoso/PS-8151-5.7

PS-8151 - Add clang-14 testing to Azure pipelines

ldonoso committed:
Merge pull request #4720 from ldonoso/PS-8151-5.7

PS-8151 - Fix warnings when compiling with clang 14

Commit:
PS-8143: Fix memory leak in File_query_log::set_rotated_name()

https://jira.percona.com/browse/PS-8143

The opt_slow_logname now uses allocated memory to keep file path.
Make sure previously allocated memory is freed when generating
rotated file name. Also add PREALLOCATED flag in variable declaration.

altmannmarcelo committed:
Merge pull request #4697 from altmannmarcelo/PS-8136-5.7

Fixed PS-8136 - LOCK TABLES FOR BACKUP is not preventing ALTER INSTAN…

Commit:
Implemented PS-8178 (Merge MySQL 5.7.38) (docs)

https://jira.percona.com/browse/PS-8178
***
Updated man pages from MySQL Server 5.7.38 source tarball.
***
Updated 'scripts/fill_help_tables.sql' from MySQL Server 5.7.38 source tarball.

Commit:
Implemented PS-8178 (Merge MySQL 5.7.38 up to 602492e78b2) (merge with conflicts)

https://jira.percona.com/browse/PS-8178

Commit:
Implemented PS-8178 (Merge MySQL 5.7.38 up to 602492e78b2) (conflicts resolved)

https://jira.percona.com/browse/PS-8178
***
VERSION raised to "5.7.38-41.0".
PERCONA_INNODB_VERSION in univ.i raised to "41.0".
***
PS related fix https://github.com/percona/percona-server/commit/81e85ac509b9c59848af1f2ac1f06227d9d61f0e
conflicts with upstream change https://github.com/mysql/mysql-server/commit/4e3e5030a27f9c75a52c546a961aafddb11571c6.
The fix was improved to be compatible with upstream code.
***
The myisam_repair_threads is deprecated after changes in
https://github.com/mysql/mysql-server/commit/b7d53db7836c0f5db3a3419727246426bd10df82.
Re-record percona_statement_set to take into account this change.
***
Re-record innodb.alter_table_stage_progress test after changes in
https://github.com/percona/percona-server/commit/963a49140c7e01cdb39bb98059951e24a9bf4269.
***
Re-record newly added main.mysqldump_bug30042589 test to include PS
specific statements from
https://github.com/percona/percona-server/commit/131a6a47afe540d3b48d7f00a84e946c5fe1f77d
and https://github.com/percona/percona-server/commit/d2d17dd676a2d68cb579b882c15d9ff817e9fccf.
***
Fix tokudb.bugs.5733_innodb and tokudb.bugs.5733_tokudb tests.
Test should mask "rows" column in explain output. Masked column number
was not updated after extending explain output some time ago.
***
The not_min_protocol_tlsv12.inc improved to directly check that TLSv1 is
allowed as this check is used to skip tests which require TLSv1.
***
Fix expected error code in group_replication.gr_recovery_connection_options test.

oleksandr-kachan committed:
Merge pull request #4722 from oleksandr-kachan/PS-8143-5.7

PS-8143: Fix memory leak in File_query_log::set_rotated_name() (5.7)

Commit:
Implemented PS-8178 (Merge MySQL 5.7.38) (no conflicts)

https://jira.percona.com/browse/PS-8178

oleksandr-kachan committed:
Merge pull request #4739 from oleksandr-kachan/merge-mysql-5.7.38-part1

PS-8178: Merge MySQL 5.7.38 part 1 (up to 602492e78b2)

oleksandr-kachan committed:
Merge pull request #4747 from oleksandr-kachan/merge-mysql-5.7.38-part2

PS-8178: Merge MySQL 5.7.38 part 2

kamil-holubicki committed:
PS-8205: Fix use of DICT_TF2_FLAG_IS_SET

https://jira.percona.com/browse/PS-8205

Problem:
During the merge it was disccovered that DICT_TF2_FLAG_SET was used
instead of DICT_TF2_FLAG_IS_SET.

Solution:
Macro usage fixed. No additional adjustments needed.
Flag is set in
create_table_info_t::enable_master_key_encryption() during table
creation.

Commit:
PS-8204: Fix XML escape rules for audit plugin

https://jira.percona.com/browse/PS-8204

There was a wrong length specified for some XML
escape rules. As a result of this terminating null symbol from
replacement rule was copied into resulting string. This lead to
quer text truncation in audit log file.
In addition added empty replacement rules for '\b' and 'f' symbols
which just remove them from resulting string. These symboles are
not supported in XML 1.0.

(cherry picked from commit c2e8878750c983446171c1b0c948029db7c566ff)

kamil-holubicki committed:
Merge pull request #4771 from kamil-holubicki/PS-8205-5.7

PS-8205: Fix use of DICT_TF2_FLAG_IS_SET

oleksandr-kachan committed:
Merge pull request #4801 from oleksandr-kachan/PS-8204-5.7.38

PS-8204: Fix XML escape rules for audit plugin

adivinho committed:
PS-8159 ps 5.7.37-40-1.buster install stucked

adivinho committed:
Merge pull request #4803 from adivinho/release-5.7.38-41

PS-8159 ps 5.7.37-40-1.buster install stucked

Commit:
Implemented PXC-3942 (Merge PS-5.7.38, Part 1 with conflicts)

https://jira.percona.com/browse/PXC-3942

Merge commit '39cdc5672f9c5c3cd5c9a476c441da0076be2225' into merge5738-part1

Conflicts:
        MYSQL_VERSION
        doc/source/conf.py
        doc/source/flexibility/binlogging_replication_improvements.rst
        doc/source/index.rst
        doc/source/installation/apt_repo.rst
        doc/source/installation/yum_repo.rst
        doc/source/management/audit_log_plugin.rst
        doc/source/myrocks/variables.rst
        doc/source/performance/threadpool.rst
        doc/source/release-notes/release-notes_index.rst
        doc/source/scalability/innodb_io.rst
        doc/source/security/data-masking.rst

***
Changes in the `doc` and `build-ps` directories were ignored and the PXC
version of them were restored.

Commit:
Implemented PXC-3942 (Merge PS-5.7.38, Part 1 conflicts resolved)

https://jira.percona.com/browse/PXC-3942

Merge conflicts resolved.

***
Bumped PXC_VERSION_EXTRA to 53.

Commit:
Implemented PXC-3942 (Merge PS-5.7.38, Part 2 no conflicts)

https://jira.percona.com/browse/PXC-3942

Merge conflicts resolved.

Percona Server release 5.7.38-41

Merge tag 'Percona-Server-5.7.38-41' into merge5738-part2

it-percona-cla commented:
[![CLA assistant check](https://cla-assistant.percona.com:443/pull/badge/not_signed)](https://cla-assistant.percona.com:443/percona/percona-xtradb-cluster?pullRequest=1646) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.percona.com:443/percona/percona-xtradb-cluster?pullRequest=1646) before we can accept your contribution.<br/>**8** out of **19** committers have signed the CLA.<br/><br/>:white_check_mark: patrickbirch<br/>:white_check_mark: altmannmarcelo<br/>:white_check_mark: venkatesh-prasad-v<br/>:white_check_mark: kamil-holubicki<br/>:white_check_mark: dutow<br/>:white_check_mark: EvgeniyPatlan<br/>:white_check_mark: alina-derkach-oaza<br/>:white_check_mark: adivinho<br/>:x: Miroslav Rajcic<br/>:x: sachinagarwal1111<br/>:x: bkandasa<br/>:x: lipengchong<br/>:x: gkodinov<br/>:x: nacarvalho<br/>:x: salmankhan123417<br/>:x: harinvadodaria<br/>:x: trosten<br/>:x: ldonoso<br/>:x: oleksandr-kachan<br/><hr/>**Miroslav Rajcic** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.percona.com:443/check/percona/percona-xtradb-cluster?pullRequest=1646) it.</sub>

kamil-holubicki reviewed (approved):
lgtm

Commit:
Implemented PXC-3942 (Merge PS-5.7.38, Part 2 Fix test failures)

Problem
-------
The testcase `galera.galera_fk_cascade_delete` failed with assertion

         rec_get_trx_id(rec, index) == thr_get_trx(thr)->id
         in btr0cur.cc

This assertion happened when two dependent cascade delete operations
were run in parallel.

Analysis
--------
Upstream fixed this issue as part of the bugfix
MDEV-26803 PA unsafety with FK cascade delete operation
(commit id: 3e3d286ef669d06e858981029c9188bc0a264251),

but the code changes were not fully integrated into PXC-5.7 during the
previous merge causing the testcase to fail.

Solution
--------
Fixed the mis-merge and now PXC marks transactions as unsafe for
parallel applying for all cascade delete operations.

Commit:
Implemented PXC-3942 (Merge PS-5.7.38, Update galera submodule pointer)

Update galera submodule pointer to include changes from upstream's
galera 25.3.37.

kamil-holubicki reviewed (approved):
lgtm
